tiledb/sm/fragment/fragment_metadata.h

```diff
@@ -698,6 +712,13 @@ class FragmentMetadata {
    */
   tuple<Status, optional<uint64_t>> get_null_count(const std::string& name);
 
+  /**
+   * Retrieves the processed conditions
+   *
+   * @return Processed conditions.
+   */
+  std::unordered_set<std::string>& get_processed_conditions();
```

Comment:
Is it correct that the order does not matter because ... we only need to check whether a .del file is in the already-processed-set?

Reply:
We will be processing all conditions for the whole array (deletes/updates) one by one, this list will be used to determine if we skip the processing for a tile, by looking in this set to see if the condition was already processed for that tile. The set allows a constant time lookup.